applications/OptimizationApplication/custom_conditions/helmholtz_surface_shape_condition.cpp

```diff
@@ -0,0 +1,511 @@
+/***********************************************************************************/
+
+Condition::Pointer HelmholtzSurfaceShapeCondition::Create(IndexType NewId, GeometryType::Pointer pGeom,  PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<HelmholtzSurfaceShapeCondition>(NewId, pGeom, pProperties);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+HelmholtzSurfaceShapeCondition::~HelmholtzSurfaceShapeCondition()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Condition::Pointer HelmholtzSurfaceShapeCondition::Clone (
+    IndexType NewId,
+    NodesArrayType const& rThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    HelmholtzSurfaceShapeCondition::Pointer p_new_cond = Kratos::make_intrusive<HelmholtzSurfaceShapeCondition>(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+    p_new_cond->SetData(this->GetData());
+    p_new_cond->Set(Flags(*this));
+
+    return p_new_cond;
+
+    KRATOS_CATCH("");
+}
+
+//************************************************************************************
+//************************************************************************************
+void HelmholtzSurfaceShapeCondition::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                            VectorType& rRightHandSideVector,
+                                            const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF_NOT(rCurrentProcessInfo.Has(COMPUTE_HELMHOLTZ_INVERSE))
+    << "COMPUTE_HELMHOLTZ_INVERSE not defined in the ProcessInfo!" << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rCurrentProcessInfo.Has(HELMHOLTZ_INTEGRATED_FIELD))
+    << "HELMHOLTZ_INTEGRATED_FIELD not defined in the ProcessInfo!" << std::endl;
+
+    KRATOS_ERROR_IF_NOT(rCurrentProcessInfo.Has(HELMHOLTZ_RADIUS))
+    << "HELMHOLTZ_RADIUS not defined in the ProcessInfo!" << std::endl;
+
+    // Check that parents have been computed
+    // These are required to retrieve the material properties and the viscous stress
+    auto& parentElement = this->GetValue(NEIGHBOUR_ELEMENTS);
+    KRATOS_ERROR_IF(parentElement.size() > 1) << "A condition was assigned more than one parent element." << std::endl;
+    KRATOS_ERROR_IF(parentElement.size() == 0) << "A condition was NOT assigned a parent element." << std::endl;
+
+    auto& r_geometry = this->GetGeometry();
+    const SizeType number_of_nodes = r_geometry.size();
+    const SizeType dimension = r_geometry.WorkingSpaceDimension();
+
+    // Resizing as needed the LHS
+    const SizeType mat_size = number_of_nodes * dimension;
+
+    if ( rLeftHandSideMatrix.size1() != mat_size )
+        rLeftHandSideMatrix.resize( mat_size, mat_size, false );
```

Comment:
```suggestion
    if ( rLeftHandSideMatrix.size1() != mat_size ) {
    rLeftHandSideMatrix.resize( mat_size, mat_size, false );
    }
```

Kratos Syleguide, Please don't use the suggestion, it kills the tabbing.